src/components/modal.js

```diff
@@ -0,0 +1,166 @@
+(() => ({
+  name: 'Modal',
+  type: 'CONTAINER_COMPONENT',
+  allowedTypes: ['BODY_COMPONENT', 'CONTENT_COMPONENT'],
+  orientation: 'HORIZONTAL',
+  jsx: (
+    <div className={B.env === 'dev' ? classes.panel : ''}>
+      {(() => {
+        const { env } = B;
+        const { Modal } = window.MaterialUI.Core;
+
+        const isEmpty = children.length === 0;
+        const isDev = env === 'dev';
+        const isPristine = isEmpty && isDev;
+
+        const [open, setOpen] = React.useState(false);
+
+        useEffect(() => {
+          B.defineFunction('handleOpenModal', e => {
+            e.preventDefault();
+
+            setOpen(true);
+          });
+
+          B.defineFunction('handleCloseModal', e => {
+            e.preventDefault();
+
+            setOpen(false);
+          });
+        }, []);
+
+        return (
+          <div>
+            {isDev && options.isVisibleInDev ? (
```

Comment:
This `invisibleInDev` option hides everything from here, but it doesn't hide the styling that comes with the className `panel` in the wrapping DIV, so you will still see something in the pagebuilder.